frame/contracts/src/benchmarking/mod.rs

```diff
@@ -2055,6 +2055,59 @@ benchmarks! {
 		let origin = RawOrigin::Signed(instance.caller.clone());
 	}: call(origin, instance.addr, 0u32.into(), Weight::MAX, None, vec![])
 
+	seal_reentrant_count {
+		let r in 0 .. API_BENCHMARK_BATCHES;
+		let code = WasmModule::<T>::from(ModuleDefinition {
+			memory: Some(ImportedMemory::max::<T>()),
+			imported_functions: vec![ImportedFunction {
+				module: "__unstable__",
+				name: "seal_reentrant_count",
+				params: vec![],
+				return_type: Some(ValueType::I32),
+			}],
+			call_body: Some(body::repeated(r * API_BENCHMARK_BATCH_SIZE, &[
```

Comment:
I think the right will be to calculate how much we need gas for one iteration over frames. And in the functions charge exactly for the count of frames. But we can leave it for official review from the `contract-pallet` team if you wish=)

Reply:
Good idea actually. I propose to move forward with it as it is for now.